src/prognostic_equations/vertical_diffusion_boundary_layer.jl

```diff
@@ -77,23 +84,47 @@ function vertical_diffusion_boundary_layer_tendency!(
 )
     ᶜρ = Y.c.ρ
     FT = Spaces.undertype(axes(ᶜρ))
-    (; ᶜp, ᶜspecific, sfc_conditions) = p.precomputed # assume ᶜts and ᶜp have been updated
-    (; C_E) = p.atmos.vert_diff
+    (; params) = p
+    thermo_params = CAP.thermodynamics_params(params)
+    (; ᶜts, ᶜp, ᶜspecific, sfc_conditions) = p.precomputed # assume ᶜts and ᶜp have been updated
+    
+    z₀ = FT(3.21e-5)
+    f_b = FT(0.1)
+    κ = FT(0.4)
+    Ri_c = FT(1)
 
     ᶠgradᵥ = Operators.GradientC2F() # apply BCs to ᶜdivᵥ, which wraps ᶠgradᵥ
+    
+    ᶠρK_E = p.scratch.ᶠtemp_scalar
+    Ri = θ_v = p.scratch.ᶜtemp_scalar
+    grav = FT(9.81)
 
+    # Compute boundary layer height
     FT = eltype(Y)
+    ᶜts_sfc = sfc_conditions.ts
+    ᶜz = Fields.coordinate_field(Y.c).z
+    zₐ = Fields.level(ᶜz, 1)
+    @. θ_v = TD.virtual_pottemp(thermo_params, ᶜts)
     interior_uₕ = Fields.level(Y.c.uₕ, 1)
-    ᶠp = ᶠρK_E = p.scratch.ᶠtemp_scalar
-    @. ᶠp[colidx] = ᶠinterp(ᶜp[colidx])
+    θ_a = Fields.level(deepcopy(θ_v), 1)
+    θ_v_sfc = @. TD.virtual_pottemp(thermo_params, ᶜts_sfc)
     ᶜΔz_surface = Fields.Δz_field(interior_uₕ)
+
+    ### Detect 𝒽, boundary layer height per column.
+    @. Ri[colidx] = compute_bulk_richardson_number(θ_v[colidx], θ_a[colidx], norm(interior_uₕ[colidx]), grav, ᶜz[colidx])
```

Comment:
I think what might happen is that Ri is 0 for some colidx, and θ_v is overwritten as they point to the same ᶜtemp_scalar. It may not be a problem as θ_v is not used after this, but if you may see these zero values if you print θ_v after this.